storage/common/include/algebra/storage/vector.hpp

```diff
@@ -0,0 +1,243 @@
+                  (!std::conjunction_v<std::is_same<array_type, Values>...>)) {
+      zero_fill(std::make_index_sequence<N - sizeof...(Values)>{});
+    }
+  }
+
+  /// Construct from existing array storage @param vals .
+  constexpr vector(const array_type &vals) : m_data{vals} {}
+
+  /// Assignment operator from wrapped data.
+  ///
+  /// @param lhs wrap a copy of this data.
+  constexpr const vector &operator=(const array_type &lhs) {
+    m_data = lhs;
+    return *this;
+  }
+
+  /// Assignment operator from @c std::initializer_list .
+  ///
+  /// @param list wrap an array of this data.
+  constexpr vector &operator=(std::initializer_list<value_type> &list) {
+    m_data = array_type(list);
+    return *this;
+  }
+
+  /// Conversion operator from wrapper to underlying data array.
+  /// @{
+  constexpr operator array_type &() { return m_data; }
+  constexpr operator const array_type &() const { return m_data; }
+  /// @}
+
+  /// Subscript operator[]
+  /// @{
+  constexpr decltype(auto) operator[](std::size_t i) { return m_data[i]; }
+  constexpr decltype(auto) operator[](std::size_t i) const { return m_data[i]; }
+  /// @}
+
+  /// Operator*=
+  ///
+  /// @return Vector expression/return type according to the operation.
+  constexpr decltype(auto) operator*=(value_type factor) noexcept {
+    return m_data *= factor;
+  }
+
+  /// Equality operators
+  /// @{
+  template <std::size_t M, typename o_value_t,
+            template <typename, std::size_t> class o_array_t,
+            template <typename, std::size_t> class p_array_t>
+  friend constexpr bool operator==(
+      const vector<M, o_value_t, o_array_t> &,
+      const vector<M, o_value_t, p_array_t> &) noexcept;
+
+  template <std::size_t M, typename o_value_t,
+            template <typename, std::size_t> class o_array_t,
+            template <typename, std::size_t> class p_array_t, bool>
+  friend constexpr bool operator==(const vector<M, o_value_t, o_array_t> &,
+                                   const p_array_t<o_value_t, M> &) noexcept;
+  /// @}
+
+  /// Inequality operator
+  template <typename other_type>
+  constexpr bool operator!=(const other_type &rhs) const noexcept {
+    return ((*this == rhs) == false);
+  }
+
+  /// Elementwise comparison. Can result in a vector-of-masks for SoA vectors
+  template <typename other_type>
+  constexpr auto compare(const other_type &rhs) const noexcept {
+    using result_t = decltype(m_data[0] == rhs[0]);
+
+    array_t<result_t, N> comp;
+
+    for (unsigned int i{0u}; i < N; ++i) {
+      comp[i] = (m_data[i] == rhs[i]);
+    }
+
+    return comp;
+  }
+
+  /// Holds the data value for every vector element
+  array_t<value_t, N> m_data;
+
+ private:
+  /// Sets the trailing uninitialized values to zero.
+  template <std::size_t... Is>
+  constexpr void zero_fill(std::index_sequence<Is...>) noexcept {
+    if constexpr (sizeof...(Is) > 0) {
+      //((m_data[N - sizeof...(Is) + Is] = value_t(0)), ...);
```

Comment:
¿que?

Reply:
This is meant to fill the rest of the vector with zeros in case too few arguments are provided (like ```vc_array4``` does). Problem being that Vc seems to interpret the comma operator as some sort of initialization overload on its ```Vc::Vector``` type and warns that this is deprecated, which becomes an error in the CI. Something fishy is going on here, but I only need it for the ```Vc::SimdArray``` based plugin later and I don't want to come up with the thing from scratch again, so I commented it for now. Open to suggestions here :)